### PR TITLE
feat: model view tabs

### LINKED ISSUE #31: [FEAT]: Add view/pages Tabs capability
**Is your feature request related to a problem? Please describe.**
* Allow the user to open multiple tabs within the app.

**Describe the solution you'd like**

Image

### PR DESCRIPTION
**Pull request by echl**
# Description

- Allow the user to open multiple tabs within the app.
- `TabContainer` component can be used to render any component as tabs.
- Refactor `Model.vue` into `ModelView` (view component that uses `TabContainer`) and `Model` (component that will be rendered in tabs)

https://user-images.githubusercontent.com/16358728/205386025-1835aef9-c5b7-4e0b-bc46-93b01cb6e854.mov

Resolves #31 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/services/hmi-server/src/main/resources/application.properties [hunk 1]
```diff
@@ -21,10 +21,10 @@ mp.openapi.extensions.smallrye.info.description=REST endpoints for the TERArium
 ########################################################################################################################
 # Microservice configurations
 ########################################################################################################################
-data-service/mp-rest/url=http://localhost:3020
+# data-service/mp-rest/url=http://localhost:3020
 # uncomment to link to actual dataservice
-# data-service/mp-rest/url=http://localhost:8001
-xdd-document-service/mp-rest/url=https://xdd.wisc.edu
+data-service/mp-rest/url=http://localhost:8001
+xdd-document-service/mp-rest/url=https://xdd.wisc.edu/api
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/resources/application.properties:27]: You should not commit this file.

File: packages/client/hmi-client/src/types/common.ts [hunk 2]
```diff
@@ -50,3 +50,9 @@ export type SidePanelTab = {
 	isGreyscale?: string;
 	badgeCount?: number;
 };
+
+export type TabMetaContent = {
+	tabName: string;
+	tabKey: number;
+	props?: Object;
+};
```
Comment (YohannParis) [inline packages/client/hmi-client/src/types/common.ts:58]:
```suggestion
export type Tab = {
	name: string;
	key: number;
	props?: Object;
};
```

File: packages/services/hmi-server/src/main/resources/application.properties (hunk 1 above)
Comment (YohannParis) [reply packages/services/hmi-server/src/main/resources/application.properties:27]: Also don't forget to merge `main` before making it non-draft

**Comment by mwdchang** [comment]
I am seeing a weird visual artifact. 

image

File: packages/client/hmi-client/src/components/tabs/Tab.vue [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+const props = defineProps({
+	name: {
+		type: String,
+		required: true,
+		default: 'New tab'
+	},
+	index: {
+		type: Number,
+		required: true,
+		default: 0
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/tabs/Tab.vue:14]: Aren't `default` and `required` mutually exclusive?

Comment (echl) [reply packages/client/hmi-client/src/components/tabs/Tab.vue:14]: Hmm you're right.

**Comment by echl** [comment]
> I am seeing a weird visual artifact.
> 
> <img alt="image" width="489" src="https://user-images.githubusercontent.com/1220927/205096720-eef463d2-8982-4dcb-8b44-f9e891022dc9.png"

There's some things I need to fix after merging main.

**Comment by Nithos** [comment]
Consider switching this to a `draft` PR if not ready for merging.

**Review by YohannParis — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]
LGTM

**Pull request merged**